### PR TITLE
Remove some uses of getMinimumIndexCompatibilityVersion

### PR DESCRIPTION
**Pull request by thecoop**
Remove uses of Version.getMinimumIndexCompatibilityVersion

### CODE REVIEW

File: distribution/tools/plugin-cli/src/test/java/org/elasticsearch/plugins/cli/RemovePluginActionTests.java [hunk 1]
```diff
@@ -141,14 +141,18 @@ public void testRemoveMultiple() throws Exception {
         assertRemoveCleaned(env);
     }
 
+    private static Version minimumCompatibleVersion(Version v) {
+        return Version.fromString((v.major - 1) + ".0.0");
+    }
+
     public void testRemoveOldVersion() throws Exception {
         Version previous = VersionUtils.getPreviousVersion();
-        if (previous.before(Version.CURRENT.minimumIndexCompatibilityVersion())) {
```
Comment (thecoop) [inline distribution/tools/plugin-cli/src/test/java/org/elasticsearch/plugins/cli/RemovePluginActionTests.java:146]: I don't think plugins actually need to reference index compatibility here?

File: server/src/test/java/org/elasticsearch/cluster/routing/allocation/FailedShardsRoutingTests.java [hunk 2]
```diff
@@ -717,13 +717,13 @@ public void testReplicaOnNewestVersionIsPromoted() {
                     .add(
                         newNode(
                             "node3-old",
-                            VersionUtils.randomVersionBetween(random(), Version.CURRENT.minimumIndexCompatibilityVersion(), null)
+                            VersionUtils.randomVersionBetween(random(), Version.CURRENT.minimumCompatibilityVersion(), null)
```
Comment (rjernst) [inline server/src/test/java/org/elasticsearch/cluster/routing/allocation/FailedShardsRoutingTests.java:720]: This doesn't seem equivalent. This test seems to want to randomize back to the oldest supported node version, as far as index compatibility is concerned, not transport (which is what minimumCompatibilityVerison is right? or at least that's what it was before creating TransportVersion)

File: distribution/tools/plugin-cli/src/test/java/org/elasticsearch/plugins/cli/RemovePluginActionTests.java (hunk 1 above)
Comment (rjernst) [reply distribution/tools/plugin-cli/src/test/java/org/elasticsearch/plugins/cli/RemovePluginActionTests.java:146]: The test is trying to mimic a plugin that was built against the oldest version that supports a full cluster restart. This if statement is for an edge case when we have just branched a new major version. In that case, calling VersionUtils.getPreviousVersion will return an unreleased version (the unreleased N.last).

In any case, I think this test is rather silly. Nothing about removing a plugin cares about the version of Elasticsearch the plugin was built for. I suggest to just remove the test entirely.

**Review by rjernst — COMMENTED** [review]
A couple thoughts

File: distribution/tools/plugin-cli/src/test/java/org/elasticsearch/plugins/cli/RemovePluginActionTests.java (hunk 1 above)
Comment (thecoop) [reply distribution/tools/plugin-cli/src/test/java/org/elasticsearch/plugins/cli/RemovePluginActionTests.java:146]: Removed

File: server/src/test/java/org/elasticsearch/cluster/routing/allocation/FailedShardsRoutingTests.java (hunk 2 above)
Comment (thecoop) [reply server/src/test/java/org/elasticsearch/cluster/routing/allocation/FailedShardsRoutingTests.java:720]: But the nodes can only form a cluster if its at least `minimumCompatibleVersion`, no?

Comment (thecoop) [reply server/src/test/java/org/elasticsearch/cluster/routing/allocation/FailedShardsRoutingTests.java]: I've modified it to specify the node & index versions separately

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by thecoop** [comment]
@elasticmachine update branch

**Pull request merged**